ddtrace/tracer/tracer_test.go

```diff
@@ -1038,11 +1040,72 @@ func TestTracerReportsHostname(t *testing.T) {
 		assert.True(ok)
 		assert.Equal(name, tracer.hostname)
 
+		name, ok = child.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(name, tracer.hostname)
+	})
+
+	t.Run("DD_TRACE_REPORT_HOSTNAME/false", func(t *testing.T) {
+		tracer, _, _, stop := startTestTracer(t)
+		defer stop()
+
+		root := tracer.StartSpan("root").(*span)
+		child := tracer.StartSpan("child", ChildOf(root.Context())).(*span)
+		child.Finish()
+		root.Finish()
+
+		assert := assert.New(t)
+
+		_, ok := root.Meta[keyHostname]
+		assert.False(ok)
 		_, ok = child.Meta[keyHostname]
 		assert.False(ok)
 	})
 
-	t.Run("disabled", func(t *testing.T) {
+	t.Run("WithHostname", func(t *testing.T) {
+		tracer, _, _, stop := startTestTracer(t, WithHostname(hostname))
+		defer stop()
+
+		root := tracer.StartSpan("root").(*span)
+		child := tracer.StartSpan("child", ChildOf(root.Context())).(*span)
+		child.Finish()
+		root.Finish()
+
+		assert := assert.New(t)
+
+		got, ok := root.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(got, hostname)
+
+		got, ok = child.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(got, hostname)
+	})
+
+	t.Run("DD_TRACE_SOURCE_HOSTNAME/set", func(t *testing.T) {
```

Comment:
We can't use `set` and `unset` here and `true` and `false` above. Chose one and let's have that in both places 🙂

Reply:
Going with set/unset